jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -554,6 +554,11 @@ YUI.add('deployment-flow', function() {
               validate={[{
                 regex: /\S+/,
                 error: 'This field is required.'
+              }, {
+                regex: /^[a-z0-9]([a-z0-9.-]*[a-z0-9])?$/,
```

Comment:
Maybe add a commenti pointing to where you took this regexp?
I'd assume https://github.com/juju/names/blob/master/model.go#L20